nbodykit/algorithms/paircount_tpcf/tpcf.py

```diff
@@ -187,7 +199,56 @@ def load(cls, output, comm=None):
         self.__setstate__(state)
         self.comm = comm
         return self
+        
+    def to_xil(self, ells, mu_range=None, mu_sel=None, return_mu_sel=False):
+        r"""
+        Invert the measured wedges :math:`\xi(r,mu)` into correlation
+        multipoles, :math:`\xi_\ell(r)`.
 
+        Parameters
+        ----------
+        ells : array_like
+            the list of multipoles to compute
+        mu_range: array_like, optional
+            the range of :math:`\mu` to use to calculate multipoles; only bins whose edges are
+            within mu_range[0] - mu-range[1] are used; if not provided, all bins are used
+        mu_sel : array_like, optional
+            the indices of bins to use to calculate multipoles;
+            if not provided, all bins within mu_range (if provided; else all bins) are used
+        return_mu_sel : boolean
+            if True, returns the indices of the :math:`\mu`-bins used to calculate multipoles
+
+        Returns
+        -------
+        xil : BinnedStatistic
+            a data set holding the :math:`\xi_\ell(r)` multipoles
+        """
+        from scipy.special import legendre
+        from scipy.integrate import quad
+
+        # new data array
+        x = str(self.corr.dims[0])
+        dtype = numpy.dtype([(x, 'f8')] + [('corr_%d' %ell, 'f8') for ell in ells])
+        data = numpy.zeros((self.corr.shape[0]), dtype=dtype)
+        dims = [x]
+        edges = [self.corr.edges[x]]
+        
+        if mu_sel is None:
+            if mu_range: mu_sel = numpy.nonzero((self.corr.edges['mu'][:-1]>=mu_range[0]) & (self.corr.edges['mu'][1:]<=mu_range[1]))[0]
+            else: mu_sel = slice(len(self.corr.edges['mu'])-1)
+        
```

Comment:
We can slice the BinnedStatistic option by `mu` using the `.sel()` function of `self.corr` here. Something like:

```python
sliced = self.corr.sel(mu=slice(*mu_range), method='nearest')
```

Reply:
I agree we could use that. Actually, I wanted to be conservative in the selected mu bins, i.e., if bin edges are [0.1,0.2] and mu_range starts at 0.18, I wanted that bin to be excluded from the calculation (to avoid polluting the result with systematics).
But I agree your solution would suit to most people.